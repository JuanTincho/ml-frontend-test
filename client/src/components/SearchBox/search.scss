@import "../../styles/colors.scss";

button {
  background: 9px 12px url(../../images/ic_Search.png) $color-gray-light;
  background-repeat: no-repeat;
  height: 42px;
  width: 35px;
}
