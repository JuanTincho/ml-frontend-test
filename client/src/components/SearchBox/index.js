import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './search.scss';

class SearchBox extends Component {
  render() {
    return (
      <div>
        <input />
        <button />
      </div>
    );
  }
}

SearchBox.propTypes = {};

export default SearchBox;
