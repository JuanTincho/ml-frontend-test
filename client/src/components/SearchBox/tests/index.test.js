import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import SearchBox from '../index';

describe('SearchBox', () => {
  let memoryRouterComponent;
  let memoryRouterInstance;
  let searchBoxInstance;
  const mockedValue = 'mockedValue';

  beforeEach(() => {
    memoryRouterComponent = mount(
      <MemoryRouter>
        <SearchBox />
      </MemoryRouter>,
    );

    memoryRouterInstance = memoryRouterComponent.instance();
    searchBoxInstance = memoryRouterComponent.find('SearchBox').instance();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const tree = renderer.create(memoryRouterComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handleChange should change the textValue of the state', () => {
    const mockedEvent = { target: { value: mockedValue } };

    expect(searchBoxInstance.state.textValue).toEqual('');
    searchBoxInstance.handleChange(mockedEvent);
    expect(searchBoxInstance.state.textValue).toEqual(mockedValue);
  });

  describe('handleKeyPress', () => {
    beforeEach(() => {
      searchBoxInstance.searchItem = jest.fn();
    });

    it('should call to searchItem when Enter is pressed', () => {
      const mockedEvent = { which: 13 };

      searchBoxInstance.handleKeyPress(mockedEvent);
      expect(searchBoxInstance.searchItem).toHaveBeenCalledTimes(1);
    });

    it('should do nothing if Enter is not pressed', () => {
      const mockedEvent = { which: 99 };

      searchBoxInstance.handleKeyPress(mockedEvent);
      expect(searchBoxInstance.searchItem).not.toHaveBeenCalled();
    });
  });

  it('goHome should set textValue to \'\' and call history.push', () => {
    // GIVEN

    // Set history.location of MemoryRouter to /search, thus later check if it changed
    memoryRouterInstance.history.push('/search');
    // Check if it changed
    expect(memoryRouterInstance.history.location.pathname).toEqual('/search');

    searchBoxInstance.setState = jest.fn();

    // WHEN
    searchBoxInstance.goHome();

    // THEN
    expect(searchBoxInstance.setState).toHaveBeenCalledTimes(1);

    // Capture the calls of setState
    const [[firstArgument, secondArgument]] = searchBoxInstance.setState.mock.calls;

    expect(firstArgument).toEqual({ textValue: '' });

    // secondArgument is an anonymus function, call it to see what it does
    secondArgument();

    // Check if location of MemoryRouter was changed by secondArgument
    expect(memoryRouterInstance.history.location.pathname).toEqual('/');
  });

  it('searchItem should set location to \'/items?search=this.state.textValue\'', () => {
    const expectedValue = `/items?search=${mockedValue}`;
    // GIVEN
    searchBoxInstance.setState({ textValue: mockedValue });

    // WHEN
    searchBoxInstance.searchItem();

    const { pathname } = memoryRouterInstance.history.location;
    const { search } = memoryRouterInstance.history.location;

    // THEN
    expect(pathname.concat(search)).toEqual(expectedValue);
  });
});
