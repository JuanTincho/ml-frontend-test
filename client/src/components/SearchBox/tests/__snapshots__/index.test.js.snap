// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly 1`] = `
<div
  className="search"
>
  <div
    className="column1"
  >
    <img
      alt="ML-logo"
      onClick={[Function]}
      src="test-file-stub"
    />
  </div>
  <div
    className="column2"
  >
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Nunca dejes de buscar"
      value=""
    />
    <button
      onClick={[Function]}
      type="submit"
    />
  </div>
</div>
`;
